krun/platform.py

```diff
@@ -0,0 +1,129 @@
+# Various CPU related abstrations
+
+import time
+import os
+import sys
+from collections import OrderedDict
+from krun.util import fatal, collect_cmd_output
+
+
+class BasePlatform(object):
+    CPU_TEMP_MANDATORY_WAIT = 1
+    CPU_TEMP_POLL_FREQ = 5
+    CPU_TEMP_POLLS_BEFORE_MELTDOWN = 12  #  * 5 = one minute
+
+    def __init__(self):
+        self.audit = OrderedDict()
+
+    def wait_until_cpu_cool(self):
+        time.sleep(BasePlatform.CPU_TEMP_MANDATORY_WAIT)
+        msg_shown = False
+        trys = 0
+        while True:
+            cool, reason = self.has_cpu_cooled()
+            if cool:
+                break
+
+            # if we get here, CPU is too hot!
+            if not msg_shown:
```

Comment:
This message is shown when we start waiting for longer than a second. Intuition is that the user may thing krun has hung. Feedback let's them know this is supposed to happen. There is another message when we wait too long too.